### PR TITLE
`Development`: Add truncate function

### PR DESCRIPTION
**Pull request by Hialus**
Add a simple truncate function for guidance to use.

### CODE REVIEW

**Review by MichaelOwenDyer — APPROVED** [review]
Looks good :)

File: app/services/guidance_functions.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+def truncate(history: list[any], max_length: int):
+    return history[-max_length:]
```
Comment (dakennguyen) [inline app/services/guidance_functions.py:2]: pls add test for it. Can max_length be negative?

**Review by dakennguyen — CHANGES_REQUESTED** [review]
pls add test for it. Can max_length be negative?

File: app/services/guidance_functions.py (hunk 1 above)
Comment (MichaelOwenDyer) [reply app/services/guidance_functions.py:2]: Come to think of it, we could address two functionalities in one here by making max_length either positive or negative, truncating the end or truncating the beginning of an array. "truncate history 3" could return up to the first 3 and "truncate history -3" could return up to the last 3 elements.

File: app/services/guidance_wrapper.py [hunk 2]
```diff
@@ -30,6 +31,7 @@ def query(self) -> Content:
         template = guidance(self.handlebars)
         result = template(
             llm=self._get_llm(),
+            truncate=truncate,
```
Comment (dakennguyen) [inline app/services/guidance_wrapper.py:34]: what is the effect after adding this?

Comment (Hialus) [reply app/services/guidance_wrapper.py:34]: This makes the function accessible within the template.

**Review by dakennguyen — APPROVED** [review]

**Pull request merged**